sdk/containerregistry/Azure.Containers.ContainerRegistry/perf/Azure.Containers.ContainerRegistry.Perf/Scenarios/ListArtifacts.cs

```diff
@@ -0,0 +1,77 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Containers.ContainerRegistry;
+using Azure.Test.Perf;
+
+namespace Azure.Containers.ContainerRegistry.Perf
+{
+    public sealed class ListArtifacts : ContainerRegistryPerfTest<PerfOptions>
+    {
+        private readonly ContainerRegistryClient _client;
+
+        public ListArtifacts(PerfOptions options) : base(options)
+        {
+            _client = new ContainerRegistryClient(new Uri(PerfTestEnvironment.Instance.Endpoint), PerfTestEnvironment.Instance.Credential);
+        }
+
+        public override async Task GlobalSetupAsync()
+        {
+            // Global setup code that runs once at the beginning of test execution.
+            await base.GlobalSetupAsync();
+
+            string repository = $"library/node";
+            string tag = "test-perf";
+
+            await ImportImage(PerfTestEnvironment.Instance.Registry, repository, tag);
+        }
+
+        public override async Task SetupAsync()
+        {
+            // Individual test-level setup code that runs for each instance of the test.
+            await base.SetupAsync();
+        }
+
+        public override async Task CleanupAsync()
+        {
+            // Individual test-level cleanup code that runs for each instance of the test.
+            await base.CleanupAsync();
+        }
+
+        public override async Task GlobalCleanupAsync()
+        {
+            // Global cleanup code that runs once at the end of test execution.
+            await base.GlobalCleanupAsync();
+        }
+
+        public override void Run(CancellationToken cancellationToken)
+        {
+            var artifactNames = new List<string>();
+
+            var repository = _client.GetRepository($"library/node");
+            foreach (var manifest in repository.GetManifests())
+            {
+                var artifact = _client.GetArtifact($"library/node", manifest.Digest);
+                artifactNames.Add(artifact.FullyQualifiedName);
```

Comment:
Do we need to capture the results in a list? This will include the list allocations as part of the test.

Reply:
No, probably not.  I was thinking we should do something with the output, but I guess there's no real need to exercise the scenario we're interested in.